Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py

```diff
@@ -0,0 +1,446 @@
+    """
+    Getting vulnerabilities from Tenable. Polling as long as the report is not ready (status FINISHED or failed)
+    Args:
+        args: arguments from user (last_found, severity and num_assets)
+        client: Client class object.
+
+    Returns: Vulnerabilities from API.
+
+    """
+    vulnerabilities = []
+    last_found = arg_to_number(args.get('last_found'))
+    num_assets = arg_to_number(args.get('num_assets')) or 5000
+    severity = argToList(args.get('severity'))
+    export_uuid = args.get('export_uuid')
+    if not export_uuid:
+        export_uuid = client.get_export_uuid(num_assets=num_assets, last_found=last_found, severity=severity)  # type: ignore
+
+    status, chunks_available = client.get_export_status(export_uuid=export_uuid)
+    if status == 'FINISHED':
+        for chunk_id in chunks_available:
+            vulnerabilities.extend(client.download_vulnerabilities_chunk(export_uuid=export_uuid, chunk_id=chunk_id))
+        readable_output = tableToMarkdown('Vulnerabilities List:', vulnerabilities,
+                                          removeNull=True,
+                                          headerTransform=string_to_table_header)
+
+        results = CommandResults(readable_output=readable_output,
+                                 raw_response=vulnerabilities)
+        return PollResult(response=results)
+    elif status in ['CANCELLED', 'ERROR']:
+        results = CommandResults(readable_output='Export job failed',
+                                 entry_type=entryTypes['error'])
+        return PollResult(response=results)
+    else:
+        results = CommandResults(readable_output='Export job failed',
+                                 entry_type=entryTypes['error'])
+        return PollResult(continue_to_poll=True, args_for_next_run={"export_uuid": export_uuid, **args},
+                          response=results)
+
+
+''' FETCH COMMANDS '''
+
+
+def fetch_vulnerabilities(client: Client, last_run: dict, severity: List[str]):
+    """
+    Fetches vulnerabilities if job has succeeded.
+    Args:
+        last_run: last run object.
+        severity: severity of the vulnerabilities to include in the export.
+        client: Client class object.
+
+    Returns:
+        Vulnerabilities fetched from the API.
+    """
+    vulnerabilities = []
+    export_uuid = last_run.get('export_uuid')
+    last_found_use = last_run.get('last_found_use')  # last run fetch time
+    if export_uuid:
+        demisto.info(f'Got export uuid from API {export_uuid}')
+        vulnerabilities, status = try_get_chunks(client=client, export_uuid=export_uuid)
+        # set params for next run
+        if status == 'FINISHED':
+            last_run.update({'export_uuid': None})
+        elif status in ['CANCELLED', 'ERROR'] and last_found_use:
+            export_uuid = client.get_export_uuid(num_assets=5000, last_found=last_found_use, severity=severity)
+            last_run.update({'export_uuid': export_uuid})
```

Comment:
1. can we use `generate_export_uuid`?
2. what if the status will constantly be cancelled/error? can we notify to the user?

Reply:
1. in my opinion we don't need to use generate_export_uuid because we are using the last_found_use as the last_found.
2. this is not likely, the documentation of tenable states that they recommend re-run the request when an error is received. And cancel is received when the admin (in Tenable) cancel the request - also not likely to happen many times in a row.